x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/deployment/DeploymentManager.java

```diff
@@ -631,12 +631,12 @@ synchronized void forcefullyStopProcess() {
             logger.debug(() -> format("[%s] Forcefully stopping process", task.getDeploymentId()));
             prepareInternalStateForShutdown();
 
-            if (priorityProcessWorker.isShutdown()) {
-                // most likely there was a crash or exception that caused the
-                // thread to stop. Notify any waiting requests in the work queue
-                handleAlreadyShuttingDownWorker();
```

Comment:
I don't know how likely this scenario is but I suppose it's possible that after we set the `running` flag to false within the `AbstractProcessWorkerExecutorService` via the call to `shutdownNow`, an interrupt occurs while the service is completing the item it was running. In that situation I believe we won't call `notifyQueueRunnables()` because of the exception. What do you think about calling `priorityProcessWorker.notifyQueueRunnables()` before `killProcessIfPresent()` just in case? If the service already drained the queue then it shouldn't do anything.

I don't think it would hurt 🤔 

A side note, I wonder if `signalAndWaitForWorkerTermination` should call `shutdownNow()` 🤔

Reply:
Good catch on the interrupt. 

I was worried about calling `notifyQueueRunnables()` while the executor is still running. I added a call to it if the `awaitTermination` returns sucess

Reply:
Just curious, what's your concern about calling `notifyQueueRunnables()` while the executor is still running? I wonder if we should do the call right before the `killProcessIfPresent()` that way we handle the potential interrupt case here too.